File: backend/Ecommerce/src/main/java/com/Ojt/Ecommerce/dto/EventDTO.java
package com.Ojt.Ecommerce.dto;

import java.time.LocalDateTime;
import java.util.List;

import com.fasterxml.jackson.annotation.JsonFormat;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class EventDTO {
    private Long id;
    private String name;
    private String description;
    private Integer slideNo;
    @JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm")
    private LocalDateTime startDate;
    @JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm")
    private LocalDateTime endDate;
    private Integer isDefault; // Optional
    private Integer status;    // Optional
    private Long discountId;   // Optional
    private List<Long> productIds; // Optional
    private String eventImage; // Multipart filename from frontend
}

File: frontend/Ecommerce/src/app/event-dto.ts
export interface EventDTO {
    id?: number;                        
    name: string;                        
    description: string;              
    slideNo: number;                  
    startDate: string;                 
    endDate: string;                   
    isDefault?: number;                
    status?: number;                    
    discountId?: number | null;      
    productIds?: number[] | null;            
    eventImage?: string;     
  }
  